models/repo/pushmirror.go

```diff
@@ -26,6 +27,10 @@ type PushMirror struct {
 	Repo       *Repository `xorm:"-"`
 	RemoteName string
 
+	// A keypair formatted in OpenSSH format.
+	PublicKey  string
```

Comment:
This will be a `TEXT` field on MySQL which is potentially too small for an RSA key - how big is the potential representation for ED25519 keys?

Reply:
I *once* run a script with the old code in `modules/crypto/ed25519.go` to generate a few million private keys and get their max length. IIRC it was 387, so I just got at 400 to be safe. 



Public keys are: `ssh-ed25519 + base64 of the ssh's marshaled of the public key bytes + new line` And should result in a max of 81 bytes per the analysis of the code (we likely should have a test that verifies these lengths)

Reply:
Might still want to enlarge it now to accomodate future key formats. There's no need to define database fields so tightly.

Reply:
I'm not able to guess future key formats. I prefer to just limit it to Ed25519 signatures and only add other signatures scheme if there's a need for them.

Reply:
If we enlarge now, it saves us from the pain of a future migration. So, unless this is a critical performance code path where there are indexing benefits by using the smaller type, I would enlarge.

Reply:
Alright then, I give in. What would be a good size to use here?

Reply:
> This will be a `TEXT` field on MySQL which is potentially too small for an RSA key - how big is the potential representation for ED25519 keys?

Forgive an long-term lurker, but isn't the maximum size of a TEXT in MySQL 65536 ([docs](https://dev.mysql.com/doc/refman/5.6/en/storage-requirements.html)).

I just generated a 16384-bit RSA key and it's nowhere near that limit:

```
$ wc -c id_rsa id_rsa.pub
12749 id_rsa
 2802 id_rsa.pub
15551 total
```

Reply:
Correct, MySQL `TEXT` should be limited to 65535 bytes, which seems plenty. So this is LGTM.